### ISSUE TITLE
Using LocalStorage instead of Cookies for storing User preferences data

### ISSUE THREAD

**Issue by sachin10101998** [fluid-project/floeproject.org#109]
## Is your feature request related to a problem?
Cookies send server information back every time an HTTP request is made. `localStorage` is an implementation of the `Storage` Interface. Unlike with cookies, data does not have to be sent back and forth with every HTTP request. This reduces the overall traffic between the client and the server and the amount of wasted bandwidth. This is because data is stored on the user’s local disk and is not destroyed or cleared by the loss of an internet connection. Also, LocalStorage can hold up to 5MB of information. This is a whole lot more than the 4KB that cookies hold.

## Describe the solution you'd like
Storing the values of preferences of a user (related to font size, colors, etc) in `localstorage` instead of Cookies. Also, with LocalStorage we can not only store strings but also Javascript primitives and objects.
TL;DR - Cookies are smaller and send server information back with every HTTP request, while LocalStorage is larger and can hold information on the client-side.

## Describe alternative solutions you've considered
Apart from `localstorage` , `sessionstorage` can be used. However, every time a person closes a tab, the session data is erased. So, people visiting the site, again and again, would feel a hurdle to change their preferences again and again. 

## Additional context or notes
1) [Cookies vs LocalStorage](https://medium.com/swlh/cookies-vs-localstorage-whats-the-difference-d99f0eb09b44)
2) [localStorage vs sessionStorage vs Cookies – a Detailed Comparison](https://wpreset.com/localstorage-sessionstorage-cookies-detailed-comparison/)

**Comment by sachin10101998** [comment]
@jhung , @jobara is there any other data stored in those cookies or some other reason for using them which I might have missed?

**Comment by jobara** [comment]
@sachin10101998 I think the first step would be to create a new store type in Infusion ( https://github.com/fluid-project/infusion/blob/master/src/framework/preferences/js/Store.js ). Please file a ticket against infusion at https://issues.fluidproject.org. 

One benefit of the cookies is that you can do server side rendering of some of the adaptations to improve speed on page load. That's not something we're currently doing on this site though, so local storage may make more sense at the moment.

**Comment by sachin10101998** [comment]
Hey @jobara, @jhung, I raised a [ticket for this issue](https://issues.fluidproject.org/browse/FLOE-575). Should I start working on this? If yes, should we start by making a new data store type in [infusion](https://github.com/fluid-project/infusion/blob/master/src/framework/preferences/js/Store.js) and then write js to change key-value pairs as per the preferences are changed in the preferences column?

**Comment by jobara** [comment]
@sachin10101998 could you file a ticket against Infusion, https://issues.fluidproject.org/projects/FLUID/issues/FLUID-5912?filter=allopenissues, for adding a new store type. You could work on that if you like.

**Comment by sachin10101998** [comment]
@jobara , just for clarity, by raising a ticket, you mean creating a new issue for Fluid Infusion project, right?

**Comment by sachin10101998** [comment]
@jobara , I created another [issue against fusion ](https://issues.fluidproject.org/browse/FLUID-6458).  I'd start working on a draft and get it reviewed by you before starting a PR.

**Comment by greatislander** [comment]
I'm going to close this as it's been captured in [FLUID-6458](https://issues.fluidproject.org/browse/FLUID-6458) and would need to be addressed in https://github.com/fluid-project/infusion.
